rust_icu_uenum/src/lib.rs

```diff
@@ -122,6 +122,26 @@ impl Iterator for Enumeration {
     }
 }
 
+impl Enumeration {
+    /// Constructs an [Enumeration] from a raw pointer.
+    ///
+    /// **DO NOT USE THIS FUNCTION UNLESS THERE IS NO OTHER CHOICE!**
+    ///
+    /// We tried to keep this function hidden to avoid the
+    /// need to have a such a powerful unsafe function in the
+    /// public API.  It worked up to a point for free functions.
+    ///
+    /// It no longer works on high-level methods that return
+    /// enumerations, since they we'd need to depend on them to
+    /// create an [Enumeration].
```

Comment:
> since they we'd need to depend on them

Can this be rephrased?

Reply:
ah, `s/they/then/` I suppose?

Reply:
SGTM.
Also, let's add `#[doc(hidden)]`?

Reply:
Done.